athenz/src/main/java/com/linecorp/armeria/server/athenz/resource/JsonBodyFieldAthenzResourceProvider.java

```diff
@@ -0,0 +1,133 @@
+ *
+ * <p>The provider supports both simple field names and nested paths using JSON Pointer notation:
+ * <ul>
+ *   <li>Simple field: {@code "resourceId"} → {@code /resourceId}</li>
+ *   <li>Nested field: {@code "user.id"} → {@code /user/id}</li>
+ * </ul>
+ *
+ * <p>Example with simple field name:
+ * <pre>{@code
+ * AthenzService.builder(ztsBaseClient)
+ *              .resourceProvider(AthenzResourceProvider.ofJsonField("resourceId"))
+ *              .action("write")
+ *              .newDecorator();
+ * }</pre>
+ *
+ * <p>Example with JSON pointer:
+ * <pre>{@code
+ * JsonPointer pointer = JsonPointer.compile("/user/id");
+ * AthenzService.builder(ztsBaseClient)
+ *              .resourceProvider(AthenzResourceProvider.ofJsonField(pointer))
+ *              .action("write")
+ *              .newDecorator();
+ * }</pre>
+ *
+ * <p>Request body example:
+ * <pre>{@code
+ * {
+ *   "resourceId": "myResource",
+ *   "user": {
+ *     "id": "123"
+ *   }
+ * }
+ * }</pre>
+ */
+@UnstableApi
+public class JsonBodyFieldAthenzResourceProvider implements AthenzResourceProvider {
+
+    private static final Logger logger = LoggerFactory.getLogger(JsonBodyFieldAthenzResourceProvider.class);
+
+    private static final ObjectMapper mapper = JacksonUtil.newDefaultObjectMapper();
+
+    private final JsonPointer jsonPointer;
+
+    /**
+     * Creates a new instance that extracts the Athenz resource from the specified JSON field.
+     *
+     * @param jsonFieldName the name of the JSON field to extract the resource from
+     */
+    protected JsonBodyFieldAthenzResourceProvider(String jsonFieldName) {
+        requireNonNull(jsonFieldName, "jsonFieldName");
+        checkArgument(!jsonFieldName.isEmpty(), "jsonFieldName must not be empty");
+        this.jsonPointer = JsonPointer.compile('/' + jsonFieldName.replace(".", "/"));
+    }
+
+    /**
+     * Creates a new instance that extracts the Athenz resource from the specified JSON pointer.
+     *
+     * @param jsonPointer the JSON pointer to extract the resource from
+     */
+    protected JsonBodyFieldAthenzResourceProvider(JsonPointer jsonPointer) {
+        requireNonNull(jsonPointer, "jsonPointer");
+        this.jsonPointer = jsonPointer;
+    }
+
+    @Override
+    public CompletableFuture<String> provide(ServiceRequestContext ctx, HttpRequest req) {
+        final MediaType contentType = req.contentType();
+        if (contentType == null || !contentType.is(MediaType.JSON)) {
+            return CompletableFuture.completedFuture("");
+        }
+        return req.aggregate().thenApply(this::extractFieldSafely);
+    }
+
+    private String extractFieldSafely(AggregatedHttpRequest agg) {
+        try {
+            final JsonNode root = mapper.readTree(agg.contentUtf8());
+            final JsonNode node = root.at(jsonPointer);
+            if (node.isMissingNode()) {
+                logger.debug("JSON pointer '{}' not found in request body", jsonPointer);
```

Comment:
Ditto. Should we throw `AthenzResourceNotFoundException` and include the exception in the returned response body?